src/ert/_c_wrappers/job_queue/workflow_job.py

```diff
@@ -3,17 +3,29 @@
 import os
 from collections import defaultdict
 from dataclasses import dataclass
-from typing import List, Optional, Type, Union
+from typing import List, Mapping, Optional, Type, Union
 
 from ert._c_wrappers.config import ConfigParser, ContentTypeEnum
 from ert._clib.job_kw import type_from_kw
-from ert.parsing import ConfigValidationError
+from ert.parsing import (
+    ConfigValidationError,
+    Instruction,
+    SchemaItemType,
+    WorkflowJobKeys,
+    init_workflow_schema,
+    lark_parse,
+)
 
 from .ert_plugin import ErtPlugin
 from .ert_script import ErtScript
 
 ContentTypes = Union[Type[int], Type[bool], Type[float], Type[str]]
 
+USE_NEW_PARSER_BY_DEFAULT = True
+
+if "USE_NEW_ERT_PARSER" in os.environ and os.environ["USE_NEW_ERT_PARSER"] == "YES":
```

Comment:
I think this needs to reuse the "USE_OLD_PARSER_BY_DEFAULT" from `ert_config.py` or specify in the environment variable that this controls the workflow job parser: `USE_NEW_WORKFLOW_JOB_PARSER`.